### ISSUE TITLE
Use with build tools 2.2.3?

### ISSUE THREAD

**Issue by Dror-Bar** [faizalshap/react-native-otp-verify#4]
Hi,
I'm trying to use this library in my project which has build tools 2.2.3.
At first I received error "could not find method google()", but all I had to do was remove that line.
Then, it didn't recognise the method "implementation", so I just changed it to compile.

So far so good, but then it gives me the error:
"Could not find support-compat.jar (com.android.support:support-compat:27.1.1)."

I don't know how to solve this one. I managed to get rid of it by using
compile "com.google.android.gms:play-services-auth:10.0.0"
instead of 
compile "com.google.android.gms:play-services-auth:+"

But then it fails to build with errors such as "package com.google.android.gms.auth.api.phone does not exist".

Any advice?

**Comment by faizalshap** [comment]
Can you try upgrading build tools to 3.1.3 ?

**Comment by Dror-Bar** [comment]
It's an existing project with plenty of other libraries and dependencies so I'm hesitant to upgrade because it might break other things.
I will attempt this though and might also test with a new project and report back.

**Comment by Dror-Bar** [comment]
Hey faizalshap,
I made a brand new project with react native version 0.57.3 and build tools 3.1.3.
It compiled successfully.
However, I'm now facing another issue in the new project. I have this code:

```
  componentDidMount() {
    RNOtpVerify.getHash()
      .then(hash => console.log(hash))
      .catch(err => console.log(err));
  }
```
And I get this error in console.log:
`Error: Cannot convert argument of type class java.util.ArrayList`
Any idea what's causing this?
Thank you.

**Comment by Dror-Bar** [comment]
@faizalshap 
I resolved the latest issue using this stackoverflow [answer](https://stackoverflow.com/a/33709182/9173778).

I changed this in RNOtpVerifyModule.java:
```
    @ReactMethod
    public void getHash(Promise promise) {
        try {
            AppSignatureHelper helper = new AppSignatureHelper(reactContext);
            ArrayList<String> signatures = helper.getAppSignatures();
            WritableArray arr = Arguments.createArray();
            for (String s : signatures)
                arr.pushString(s);
            promise.resolve(signatures);
        } catch (Exception e) {
            promise.reject(e);
        }
    }
```
To this:
```
    @ReactMethod
    public void getHash(Promise promise) {
        try {
            AppSignatureHelper helper = new AppSignatureHelper(reactContext);
            ArrayList<String> signatures = helper.getAppSignatures();
            WritableArray arr = Arguments.createArray();
            for (String s : signatures)
                arr.pushString(s);
            promise.resolve(arr); // I'm resolving arr, which is a WritableArray instead of signatures which is ArrayList
        } catch (Exception e) {
            promise.reject(e);
        }
    }
```

**Comment by Dror-Bar** [comment]
Regarding my original issue, I was at fault.
When creating a new React Native project with version 0.55.4 (which uses build tools 2.2.3), your library compiled just fine (I still had to make the change mention above to get the hash though).

Turns out that I was using build tools 2.2.3 but changed my buildToolsVersion to 27.0.3, which seemed to have caused an issue - some sort of a mismatch with dependencies I assume. At the end I was able to fix this by upgrading to build tools 3.1.4 successfully.
